gradle/libs.versions.toml

```diff
@@ -0,0 +1,89 @@
+# which runs on JDK 11.
+layoutLibPrebuiltSha = 'fa3aa65'
+layoutlib = "2021.2.1-patch1-fa3aa65"
+
+
+guava = "31.0.1-jre"
+kxml2 = "2.3.0"
+okio = "3.1.0"
+moshi = "1.13.0"
+jcodec = "0.2.5"
+bytebuddy = "1.12.10"
+
+junit = "4.13.2"
+testParameterInjector = "1.8"
+truth = "1.1.3"
+assertj = "3.23.1"
+
+[libraries]
+layoutlib-native-jdk11 = { module = "app.cash.paparazzi:layoutlib-native-jdk11", version.ref = "layoutlib" }
+layoutlib-native-macArm = { module = "app.cash.paparazzi:layoutlib-native-macarm", version.ref = "layoutlib" }
+layoutlib-native-macOsX = { module = "app.cash.paparazzi:layoutlib-native-macosx", version.ref = "layoutlib" }
+layoutlib-native-windows = { module = "app.cash.paparazzi:layoutlib-native-win", version.ref = "layoutlib" }
+layoutlib-native-linux = { module = "app.cash.paparazzi:layoutlib-native-linux", version.ref = "layoutlib" }
+
+tools-common = { module = "com.android.tools:common", version.ref = "tools-common" }
+tools-layoutlib = { module = "com.android.tools.layoutlib:layoutlib-api", version.ref = "tools-layoutlib" }
+tools-ninepatch = { module = "com.android.tools:ninepatch", version.ref = "tools-ninepatch" }
+tools-sdkCommon = { module = "com.android.tools:sdk-common", version.ref = "tools-sdkCommon" }
+
+androidx-annotations = { module = "androidx.annotation:annotation", version.ref = "androidx-annotations" }
+androidx-compose-compiler = { module = "androidx.compose.compiler:compiler", version.ref = "androidx-compose-compiler" }
+androidx-compose-ui-material = { module = "androidx.compose.material:material", version.ref = "androidx-compose-ui-material" }
+androidx-lifecycle-common = { module = "androidx.lifecycle:lifecycle-common", version.ref = "androidx-lifecycle" }
+
+kotlin-bom = { module = "org.jetbrains.kotlin:kotlin-bom", version.ref = "kotlin" }
+kotlinx-coroutines-core = { module = "org.jetbrains.kotlinx:kotlinx-coroutines-core", version.ref = "kotlin-coroutines" }
+
+guava = { module = "com.google.guava:guava", version.ref = "guava" }
+kxml2 = { module = "kxml2:kxml2", version.ref = "kxml2" }
+okio = { module = "com.squareup.okio:okio", version.ref = "okio" }
+
+moshi-adapters = { module = "com.squareup.moshi:moshi-adapters", version.ref = "moshi" }
+moshi-core = { module = "com.squareup.moshi:moshi", version.ref = "moshi" }
+moshi-kotlin = { module = "com.squareup.moshi:moshi-kotlin-codegen", version.ref = "moshi" }
+
+jcodec-core = { module = "org.jcodec:jcodec", version.ref = "jcodec" }
+jcodec-javase = { module = "org.jcodec:jcodec-javase", version.ref = "jcodec" }
+bytebuddy-agent = { module = "net.bytebuddy:byte-buddy-agent", version.ref = "bytebuddy" }
+bytebuddy-core = { module = "net.bytebuddy:byte-buddy", version.ref = "bytebuddy" }
+
+junit = { module = "junit:junit", version.ref = "junit" }
+testParameterInjector = { module = "com.google.testparameterinjector:test-parameter-injector", version.ref = "testParameterInjector" }
```

Comment:
let's inline all versions that weren't already extracted.  we only use the extracted versions for version that appear more than once (except for agp, probably cuz i copied from a repo that pinned lint to the same version).

Reply:
I saw that, and I actually tried it myself on my projects, but it became very confusing very fast to go to different places to update the version numbers.

Having them all in one place gives a really nice/clean/quick overview of what the project uses, without all the noise of group and module names.

Version history becomes simpler, you can blame the versions block to see how things changed over time.

Also consider: "where are the versions? under `[versions]`" vs. "where are the versions? well, it depends on how many usages are there, ... long explanation."

Still happy to go ahead inlining, just wanted to give my 2 cents.

Reply:
ack.  thanks for the perspective, but yea, our convention is to inline single-used versions.